src/basic/hashmap.c

```diff
@@ -2149,6 +2151,40 @@ int _hashmap_dump_sorted(HashmapBase *h, void ***ret, size_t *ret_n) {
 
         typesafe_qsort_r(entries, n, hashmap_entry_compare, h->hash_ops->compare);
 
+        *ret = TAKE_PTR(entries);
+        *ret_n = n;
+        return 0;
+}
+
+int _hashmap_dump_keys_sorted(HashmapBase *h, void ***ret, size_t *ret_n) {
+        _cleanup_free_ struct hashmap_base_entry **entries = NULL;
+        size_t n;
+        int r;
+
+        r = _hashmap_dump_entries_sorted(h, &entries, &n);
+        if (r < 0)
+                return r;
+
+        /* Reuse the array. */
+        FOREACH_ARRAY(e, entries, n)
+                *e = (void*) (*e)->key;
```

Comment:
This MR is stacked on top of #30840 

Is there something specific I'm supposed to do so that you don't accidentally review the same things twice over?